File: Java/ArrayString/02_containerWithMaximumWater.java
public int maxArea(int[] height) {
        int n = height.length;
        int i = 0, j = n-1;
        int area = 0;
        while(i<j){
            int b = j-i;                
            area = (height[i] < height[j]) ?  Math.max(area, height[i++]*b) : Math.max(area, height[j--]*b);  
            // area = Math.max(area, (Math.min(height[i],height[j])) * (j-i));
            // if(height[i]>height[j])  j--;
            // else i++;
        }
        return area;
    }

File: Python/ArrayString/02_containerWithMaximumWater.py
def maxArea(self, height: List[int]) -> int:
    left, right = 0, len(height) - 1;
    length, width = 0, 0

    ans = 0

    while(left < right) :
        length = right - left
        width = min(height[left], height[right])
        ans = max(ans, length * width)
        if height[left] < height[right]:
            left += 1
        else:
            right -= 1
    
    return ans